src/main/scala/org/ergoplatform/api/ScriptApiRoute.scala

```diff
@@ -91,6 +114,74 @@ case class ScriptApiRoute(readersHolder: ActorRef, ergoSettings: ErgoSettings)
     }
   }
 
+
+  case class ExecuteRequest(script: String,
+                            env: Map[String,Any],
+                            ctx: ErgoLikeContext)
+    extends TransactionRequest
+
+
+  implicit final val decodeMap: Decoder[Any] = new Decoder[Any] {
+    final def apply(c: HCursor): Result[Any] = {
+      val x = DataJsonEncoder.decode(c.value)
+      Right(x)
+    }
+  }
+
+  class ExecuteRequestDecoder(settings: ErgoSettings) extends Decoder[ExecuteRequest] with JsonCodecs {
+    def apply(cursor: HCursor): Decoder.Result[ExecuteRequest] = {
+      for {
+        script <- cursor.downField("script").as[String]
+        env <- cursor.downField("env").as[Map[String,Any]]
+        ctx <- cursor.downField("ctx").as[ErgoLikeContext]
+      } yield ExecuteRequest(script, env.asInstanceOf[Map[String,Any]], ctx)
+    }
+  }
+
+  implicit val executeRequestDecoder: ExecuteRequestDecoder = new ExecuteRequestDecoder(ergoSettings)
+
+  implicit def sigmaBooleanEncoder: Encoder[SigmaBoolean] = {
+    sigma =>
+      val op = sigma.opCode.toByte.asJson
+      sigma match {
+        case dlog: ProveDlog   => Map("op" -> op, "h"->dlog.h.asJson).asJson
+        case dht: ProveDHTuple => Map("op" -> op, "g"->dht.g.asJson, "h"->dht.h.asJson, "u"->dht.u.asJson, "v"->dht.v.asJson).asJson
+        case _: TrivialProp => Map("op" -> op).asJson // besides opCode no additional bytes
```

Comment:
@knizhnik it should also save `condition` (see `abstract class TrivialProp(val condition: Boolean)`)